client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -596,6 +596,55 @@ public void flush() throws SegmentSealedException {
         }
     }
 
+    /**
+     * @see SegmentOutputStream#flushAsync()
+     */
+    @Override
+    public CompletableFuture<Boolean> flushAsync() throws SegmentSealedException {
```

Comment:
This is really similar to flush() above. We probably want to think about a way for them to share code.